process_email.py

```diff
@@ -1028,97 +1028,179 @@ def process_email(self, base_connector, rfc822_email, email_id, config, epoch, c
+                continue
+
+            # add the vault id artifact to the container
+            cef_artifact = curr_file.get('meta_info', {})
+            if (file_name):
+                cef_artifact.update({'fileName': file_name})
+
+            if vault_id:
+                cef_artifact.update({'vaultId': vault_id,
+                    'cs6': vault_id,
+                    'cs6Label': 'Vault ID'})
+
+                # now get the rest of the hashes and add them to the cef artifact
+                self._add_vault_hashes_to_dictionary(cef_artifact, vault_id, container_id)
+
+            if (not cef_artifact):
+                continue
+
+            artifact = {}
+            artifact['name'] = 'Vault Artifact'
+            # set the artifact severity as configured in the asset, otherwise the artifact will get the default 'medium' severity
+            # The container picks up the severity of any artifact that is higher than it's own
+            artifact['severity'] = self._base_connector.get_config().get('container_severity', 'medium')
+            artifact['cef'] = cef_artifact
+
+            if (contains):
+                artifact['cef_types'] = {'vaultId': contains, 'cs6': contains}
+            self._set_sdi(artifact)
+
+            if ('parentGuid' in cef_artifact):
+                parent_guid = cef_artifact.pop('parentGuid')
+                cef_artifact['parentSourceDataIdentifier'] = self._guid_to_hash.get(parent_guid)
+                self._debug_print("The value of parentSourceDataIdentifier in cef_artifact of is: {}".format(
+                    cef_artifact.get('parentSourceDataIdentifier')))
+
+            vault_artifacts.append(artifact)
+
+        # add all the vault artifacts to the list of other artifacts ingested to save them together
+        if vault_artifacts:
+            artifacts.extend(vault_artifacts)
+
+        container['artifacts'] = artifacts
+        if (hasattr(self._base_connector, '_preprocess_container')):
+            container = self._base_connector._preprocess_container(container)
+
+        artifacts_list = container['artifacts']
+
+        for artifact in artifacts_list:
+            artifact['container_id'] = container_id
+            artifact['run_automation'] = False
+
+        if duplicate_container:
+            artifacts[-1]['run_automation'] = self._config.get('automation_on_duplicate', True)
+        else:
+            artifacts[-1]['run_automation'] = True
+
+        # save all the artifacts(Vault, IP, domain, etc.) in a single save_artifacts call
+        ret_val, message, ids = self._base_connector.save_artifacts(artifacts_list)
+        self._base_connector.debug_print(
+            "save_artifacts returns, value: {0}, reason: {1}".format(
+                ret_val,
+                message
             )
+        )
 
-            if (added_to_vault):
-                vault_artifacts_added += 1
+        if (phantom.is_fail(ret_val)):
```

Comment:
```suggestion
        if phantom.is_fail(ret_val):
```
I'm going to stop commenting these now even if I see more later lol, but same for all the other cases with unnecessary parentheses

Reply:
Removed all unnecessary parentheses